test/src/test/java/hudson/tasks/BuildTriggerTest.java

```diff
@@ -309,13 +310,6 @@ public void downstreamProjectShouldObserveCompletedParent() throws Exception {
 
         final FreeStyleProject us = j.createFreeStyleProject();
         us.getPublishersList().add(new BuildTrigger("downstream", true));
-        us.getPublishersList().add(new TestNotifier() {
-            @Override
-            public boolean perform(AbstractBuild<?, ?> build, Launcher launcher, BuildListener listener) throws InterruptedException, IOException {
-                Thread.sleep(5000); // Build does not complete just after the BuildTrigger
-                return true;
-            }
-        });
```

Comment:
Ah, did not see that you created the test in a separate commit from the fix. All clear now.